requirements.in

```diff
@@ -5,36 +5,38 @@
 # to update `requirements.txt` and `third_party/python_libs` accordingly.
 apache-beam[gcp]==2.24.0
 backports.functools-lru-cache==1.6.1
-beautifulSoup4==4.9.1
+beautifulsoup4==4.9.3
 bleach==3.3.0
 callbacks==0.3.0
 contextlib2==0.6.0.post1
-# TODO(#11474): Use a regular dependency after upgrading to Python 3. Until
-# then, we must depend on a patched version of firebase-admin==3.2.1 (hosted on
-# the following repository) because that is the last version which supports
-# Python 2, but is missing a feature which we were forced to patch in.
-git+git://github.com/oppia/firebase-admin-python@8d6deee889733d09ef5327d78c09e4d7b17785ac#egg=firebase-admin
+firebase-admin==5.0.0
 future==0.18.2
-GoogleAppEngineCloudStorageClient==1.9.22.1
-GoogleAppEngineMapReduce==1.9.22.0
+google-cloud-storage==1.37.1
 GoogleAppEnginePipeline==1.9.22.1
 Graphy==1.0.0
 html5lib==1.0.1
+mailchimp3==3.0.14
 mutagen==1.43.0
 packaging==20.4
 pylatexenc==2.6
 simplejson==3.17.0
 six==1.15.0
 soupsieve==1.9.5
 webencodings==0.5.1
+pillow==8.2.0
 redis==3.5.3
 requests==2.24.0
 requests-mock==1.5.2
 requests-toolbelt==0.9.1
+googledatastore==7.0.0
+google-auth==1.28.1
+google-cloud-ndb==1.8.0
 google-cloud-tasks==1.5.0
 google-cloud-translate==2.0.1
-googleapis_common_protos==1.52.0
+pytest==6.2.3
 webapp2==3.0.0b1
 elasticsearch==7.10.0
-mailchimp3==3.0.14
-typing==3.7.4.3
+# Libraries that are already in oppia_tools folder, but are needed for deployed
```

Comment:
What differentiates the libs below from the ones above? The ones above are in oppia_tools as well, right?

(I'm basically wondering if it makes sense to put all our python third-party libs in the same place. If not, we should have a clear understanding of what differentiates the various places we put them in.)

Reply:
You might be mistaking ../oppia_tools and third_party/python_libs. We already have these two folders now, ../oppia_tools is for tooling, third_party/python_libs is for libraries that we use in the actual server. All files in this folders are installed into third_party/python_libs, even though we have them installed in ../oppia_tools, this is because when we deploy to the production we don't have ../oppia_tools there and thus we need the libraries listed here.

Reply:
So why don't we move these two libs to third_party/python_libs instead?

Reply:
Because when we use them for tooling we need to have these as early as possible, before we start installing everything else, even the other Python lins to third_party/python_libs (see https://github.com/oppia/oppia/blob/develop/scripts/install_third_party_libs.py#L31). I think this whole library handling might be reconsidered after the migration, but I don't even have an idea on how would this go and thus problem adding TODO for this.

Reply:
OK. Explain all that stuff here, it's important context for why these two libs are treated specially which this file does not currently explain.

For filing the issue: have the issue explain the problem, but the task for the issue should be "find a solution such that constraints X and Y are satisfied".

Reply:
Done.